src/apps/tools/OswAppStepStats.cpp

```diff
@@ -43,7 +39,10 @@ void OswAppStepStats::drawInfoPanel(OswUI* ui, uint32_t pos, uint32_t lastWeekDa
     hal->gfx()->setTextCenterAligned();
     hal->gfx()->setTextBottomAligned();
     hal->gfx()->setTextCursor(DISP_W / 2, 170);
-    hal->gfx()->print(hal->getLocalWeekday(&pos));
+    OswDate oswDate = { };
+    hal->getLocalDate(oswDate);
+    const char* weekday = hal->getWeekDay[oswDate.weekDay];
```

Comment:
please use `std::array::at` instead - using the `operator[]` with an invalid index will result in an invalid memory access (as always with C-arrays), but `std::array::at` will throw an exception instead.